catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileAlterationObserver.java

```diff
@@ -137,17 +139,16 @@ public boolean checkAndNotify() {
         LOGGER.debug("Another thread is currently running, returning until next poll");
         return false;
       }
-      isProcessing = true;
-    }
 
-    //  You cannot change listeners in the middle of executions.
-    AsyncFileAlterationListener listenerCopy;
-    synchronized (listenerLock) {
-      if (listener == null) {
-        isProcessing = false;
-        return false;
+      isProcessing = true;
```

Comment:
❗️ Since `isProcessing` is read and modified outside of a single lock, it must be defined as volatile to ensure it is always up to date. Either that or it should be always read and updated under the same lock.